### PR TITLE
ignore templates.go

### PR DESCRIPTION
**Pull request by jsilberm**
**What this PR does / why we need it**:
Removes the need for including "templates.go" in future PRs where template files have changed.
This should remove much friction associated with community contributions to acs-engine.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

[878](https://github.com/Azure/acs-engine/issues/878)

**Special notes for your reviewer**:

**Release note**:

```release-note
```

---
This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/915)

### CODE REVIEW

**Comment by seanknox** [comment]
@acs-bot test this please

File: .gitignore [hunk 1]
```diff
@@ -11,4 +11,5 @@ test/acs-engine-test/acs-engine-test
 
 test/acs-engine-test/acs-engine-test.exe
 pkg/operations/junit.xml
-pkg/operations/kubernetesupgrade/junit.xml
\ No newline at end of file
+pkg/operations/kubernetesupgrade/junit.xml
+pkg/acsengine/templates.go
```
Comment (anhowe) [inline .gitignore]: If templates.go is not checked in, do we know if this breaks downstream dependencies that rely on this file to be there?  @JackQuincy

Comment (rjtsdl) [inline .gitignore]: We cannot ignore templates.go file. 

If we keep a legacy template.go file for ever, new changes into parts/ won't be taken, right?

**Comment by jsilberm** [comment]
According to @JackQuincy on the corresponding [Issues thread](https://github.com/Azure/acs-engine/issues/878),  the issue revolves around vendoring (which seems to be an internal project issue).  If you want to ellicit greater community participation and collaboration, then the project team should bias review/infrastructure/ci/etc towards that goal of greater community participation.   

I have heard from a number of folks within Microsoft that they share the view here.  If you want "acs-engine" to have wings, then you need to remove the friction.

**Comment by rjtsdl** [comment]
@jsilberm , i think i just get the idea.

For RP usage, we do post vendor go generate
For community usage, we want to include `go generate` in the makefile. 

Makes sense to me now.

**Comment by anhowe** [comment]
@jsilberm I completely agree, and we are trying to remove friction like this.

@JackQuincy are we ready to take a change like this?

**Comment by jsilberm** [comment]
Hi @rjtsdl .   Can you please share the concern between "LGTM" and "Hold on the change for now"?

**Comment by anhowe** [comment]
@jsilberm we are temporary delaying this to ensure downstream dependencies do not break.

**Comment by jsilberm** [comment]
Sounds like we're getting close to closure on this.  The ball is in @JackQuincy 's court?

**Comment by JackQuincy** [comment]
Yes @jsilberm , I'm 98% of the way there. It is ugly downstream but it should be manageable and the issues here in this repo are getting unmanageable.

**Comment by seanknox** [comment]
Thanks @JackQuincy and @jsilberm! This will help a lot of folks out.

**Pull request merged**